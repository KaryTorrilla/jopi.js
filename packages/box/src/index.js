import styled from 'styled-components'
import {
  space,
  color,
  layout,
  flexbox,
  border,
  typography,
  position,
  variant,
} from 'styled-system'
import css from '@styled-system/css'

const sx = props => css(props.sx)(props.theme)
const base = props => css(props.__css)(props.theme)
const theme = ({ tx = 'variants' }) =>
  variant({
    scale: tx,
  })

export const Box = styled.div(
  {
    boxSizing: 'border-box',
    margin: 0,
    minWidth: 0,
  },
  base,
  theme,
  sx,
  space,
  color,
  layout,
  flexbox,
  typography,
  border,
  position
)

export const Flex = styled(Box)({
  display: 'flex',
})
