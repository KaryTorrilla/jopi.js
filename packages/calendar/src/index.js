import React from 'react'
import FullCalendar from '@fullcalendar/react' // must go before plugins
import dayGridPlugin from '@fullcalendar/daygrid' // a plugin!
import '@oneloop/theme/styles/globals.css'
import '../styles/styles.css'

export const Calendar = () => {
  /*   const calendarOptions = {
      events: [],
      plugins: [timeGridPlugin],
      initialView: 'timeGridDay',
      nowIndicator: true,
      headerToolbar: {
        start: 'prev',
        center: 'title',
        end: 'next',
      },
      editable: true,
      eventResizableFromStart: true,
      dragScroll: true,
      dayMaxEvents: 2,
      navLinks: true,
    } */

  return (
    <div className='calendarContainer'>
      <FullCalendar
        plugins={[dayGridPlugin]}
        initialView="dayGridMonth"
      />
    </div>
  )
}
