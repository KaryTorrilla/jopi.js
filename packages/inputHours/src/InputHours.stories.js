import React from 'react'
import { InputHours } from '.'

export default {
  component: InputHours,
  title: 'InputHours',
  argTypes: {
    inputTime: {
      name: 'inputTime',
      description:
        'Horario inicial del input, formato day. Por defecto carga la fecha actual',
      type: 'text',
      control: { type: 'none' },
    },
    arrayInput: {
      name: 'arrayInput',
      description: 'Array de horas, por defecto carga todas las horas',
      type: 'array',
      control: { type: 'none' },
    },
    val: {
      name: 'val',
      description: 'Recibe una funciÃ³n que devuelve la fecha del input',
      type: 'function',
      control: { type: 'none' },
    },
    error: {
      name: 'error',
      description: 'Booleano para indicar error en el input',
      type: 'boolean',
      control: { type: 'none' },
    },
  },
}

export const Hours = () => <InputHours />
