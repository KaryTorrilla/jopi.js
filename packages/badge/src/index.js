import React from 'react'
import { Box } from '@oneloop/box'
import '@oneloop/fonts'

export const Badge = ({ isMedium, hasIcon, isNotButton, ...props }) => (
  <Box
    as={ isNotButton ? "" : "button" }
    tx="buttons"
    variant="primary"
    {...props}
    __css={{
      lineHeight: '14px',
      height: isMedium ? '22px' : '16px',
      display: 'inline-block',
      fontSize: '10px',
      fontFamily: 'primary',
      textAlign: 'center',
      textTransform: 'uppercase',
      fontWeight: 'bold',
      letterSpacing: '0.05em',
      px: 2,
      py: '2px',
      color: 'white',
      bg: 'primary',
      border: 0,
      borderRadius: isMedium ? '14px' : '8px',
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
      padding: hasIcon ? '4px 8px' : '1px 5px',
      position: 'static',
      flex: 'none',
      order: 0,
      flexGrow: 0,
      margin: '0px 0px',
    }}
  />
)
