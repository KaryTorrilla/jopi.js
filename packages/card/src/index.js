import React from 'react'
import { Box } from '@oneloop/box'
import theme from '@oneloop/theme'

export const Card = ({ variant, active, ...props }) => {
  const variantValues = Object.values(theme.card)[Object.keys(theme.card).indexOf(variant)]
  const backgroundColor = active ? variantValues[':focus'].backgroundColor : variantValues.backgroundColorValue
  return (
    <Box sx={{ position: 'relative' }}>
      <Box
        tx='card'
        variant={variant}
        {...props}
        __css={{
          fontFamily: 'primary',
          display: 'flex',
          flexDirection: 'row',
          alignItems: 'center',
          backgroundColor: backgroundColor,
          position: 'static',
        }}
        />
    </Box>
  )
}
