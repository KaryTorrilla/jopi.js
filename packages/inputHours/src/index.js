import React, { useEffect, useRef, useState } from 'react'
import { Box } from '@oneloop/box'
import { Icon } from '@oneloop/icons'
import { Text } from '@oneloop/text'
import { Input } from '@oneloop/input'
import { useOnClickOutside } from '@oneloop/hooks'
import { format, addMinutes, set } from 'date-fns'

const defaultHours = []

for (let i = 0; i < 24; i++) {
  for (let j = 0; j < 4; j++) {
    defaultHours.push(`${i < 10 ? '0' + i : i}:${j === 0 ? '00' : 15 * j}`)
  }
}

export const InputHours = ({
  inputTime = new Date(),
  arrayInput = defaultHours,
  variant = 'default',
  error = false,
  val = () => {},
}) => {
  const InputContRef = useRef(null)
  const listItemsRef = useRef(null)
  const [showDD, setShowDD] = useState(false)
  const [time, setTime] = useState()
  const [inputVariant, setInputVariant] = useState(variant)

  useEffect(() => {
    if (error) {
      setInputVariant('focus')
    }
  }, [error])

  useOnClickOutside(InputContRef, () => setShowDD(false))

  const handleShowDD = () => setShowDD(!showDD)

  const hoursFormatRef = useRef({ hours: 0, minutes: 0, seter: inputTime })

  const handleChangeTime = (theTime) => {
    setTime(format(theTime, 'HH:mm'))
    hoursFormatRef.current.seter = theTime
  }

  const handleArrows = (min) => {
    const { seter } = hoursFormatRef.current
    if (format(seter, 'mm') % 15) {
      handleChangeTime(set(seter, { minutes: 0 }))
    } else {
      hoursFormatRef.current = {
        ...hoursFormatRef.current,
        seter: addMinutes(seter, min),
      }
      handleChangeTime(addMinutes(seter, min))
    }
  }

  const handleInputChange = (event) => {
    const regex = /^[0-9]+$/
    const newValue = event.target.value.replace(/[^0-9]/g, '')
    if (regex.test(newValue) || event.target.value === '') {
      hoursFormatRef.current = {
        ...hoursFormatRef.current,
        hours: Number(newValue.substring(0, 2)) | 0,
        minutes: Number(newValue.substring(2, 4)) | 0,
      }
      const { hours, minutes } = hoursFormatRef.current
      if (hours >= 0 && hours <= 23 && minutes >= 0 && minutes <= 59) {
        hoursFormatRef.current = {
          ...hoursFormatRef.current,
          seter: set(inputTime, { hours: hours, minutes: minutes }),
        }
        setTime(newValue)
      }
    }
  }

  const handleBlur = () => {
    handleChangeTime(hoursFormatRef.current.seter)
    setInputVariant(error ? 'focus' : 'default')
  }

  const handleClickTime = (h) => {
    const seter = set(inputTime, {
      hours: h.substring(0, 2),
      minutes: h.substring(3, 5),
    })
    handleChangeTime(seter)
  }

  const handleInitialTime = () => {
    const restTime = format(inputTime, 'mm') % 15
    const interval = restTime < 15 ? 30 : 15

    hoursFormatRef.current = {
      ...hoursFormatRef.current,
      seter: addMinutes(inputTime, interval - restTime),
    }

    handleChangeTime(addMinutes(inputTime, interval - restTime))
  }

  const handleScroll = () => {
    const index = arrayInput.indexOf(time?.substring(0, 5)) * 26 - 52
    listItemsRef.current.scrollTop = index
  }

  useEffect(() => {
    const currentTime = format(new Date(), 'HH:mm')
    const input = format(inputTime, 'HH:mm')

    if (currentTime !== input) {
      handleClickTime(input)
    }
  }, [inputTime])

  useEffect(() => {
    handleInitialTime()
  }, [])

  useEffect(() => {
    handleScroll()
    if (time) {
      val(
        set(inputTime, {
          hours: time.slice(0, 2),
          minutes: time.slice(3, 5),
        })
      )
    }
  }, [time, listItemsRef.current])

  return (
    <Box
      tx="inputHours"
      variant={inputVariant}
      __css={{
        position: 'relative',
        width: 'fit-content',
        borderRadius: '8px',
        border: 'solid 1px transparent',
      }}
      ref={InputContRef}
      onClick={handleShowDD}
    >
      <Input
        id="inputHourDisplay"
        prefix={<Icon icon="icon-reloj" fontSize="14px" />}
        suffix={<Icon icon="icon-dropdown" fontSize="14px" />}
        variant="inputSearch"
        maxLength="4"
        onChange={handleInputChange}
        onFocus={() => setInputVariant('active')}
        onBlur={handleBlur}
        value={time}
        onKeyDown={(e) => {
          if (e.key === 'Enter') {
            e.target.blur()
            handleBlur()
            setShowDD(false)
          }
          if (e.key === 'ArrowDown') {
            handleArrows(15)
          }
          if (e.key === 'ArrowUp') {
            handleArrows(-15)
          }
        }}
      />
      <Box
        as="ul"
        ref={listItemsRef}
        className="List"
        __css={{
          width: InputContRef.current?.offsetWidth,
          zIndex: '100',
          backgroundColor: '#FFF',
          visibility: showDD ? 'visible' : 'hidden',
          position: 'absolute',
          height: '140px',
          overflow: 'auto',
          top: '35px',
          boxShadow: '4px 4px 12px rgba(87, 95, 99, 0.1)',
          borderRadius: '8px',
          padding: '0px',
          listStyle: 'none',
          overscrollBehavior: 'contain',
          '::-webkit-scrollbar': {
            width: '4px',
          },
          '::-webkit-scrollbar-track': {
            '-webkit-border-radius': '4px',
            background: 'transparent',
          },
          '::-webkit-scrollbar-thumb': {
            '-webkit-border-radius': '4px',
            background: '#6F838D',
          },
        }}
      >
        {arrayInput.map((h, i) => (
          <Box
            key={i}
            as="li"
            tx="liInputHours"
            variant={
              arrayInput.indexOf(h) === arrayInput.indexOf(time)
                ? 'selected'
                : ''
            }
            __css={{
              padding: '4px 7px',
              cursor: 'pointer',
              ':hover': {
                backgroundColor: '#F3F6F8',
                color: '#384248',
              },
              ':first-child': {
                borderRadius: '8px 8px 0px 0px',
              },
              ':last-child': {
                borderRadius: '0px 0px 8px 8px',
              },
            }}
            onClick={() => handleClickTime(h)}
          >
            <Text variant="body.fontSize14">{h}</Text>
          </Box>
        ))}
      </Box>
    </Box>
  )
}
