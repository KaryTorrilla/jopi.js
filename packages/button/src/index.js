import React from 'react'
import { Badge } from '@oneloop/badge'
import { Box } from '@oneloop/box'
import { Checkbox } from '@oneloop/checkbox'
import { Icon } from '@oneloop/icons'
import { Image } from '@oneloop/image'
import { useToggle } from '@oneloop/hooks'
import theme from '@oneloop/theme'
import '@oneloop/fonts'

export const Button = ({ variant, ...props }) => (
  <Box sx={{ position: 'relative' }}>
    <Box
      as='button'
      tx='buttons'
      variant={variant}
      {...props}
      __css={{
        appearance: 'none',
        display: 'flex',
        lineHeight: 'inherit',
        fontFamily: 'Nunito Sans',
        fontWeight: 'normal',
        textAlign: 'center',
        textDecoration: 'none',
        cursor: 'pointer',
        outline: 'none',
        color: 'white',
        bg: 'primary',
        border: 0,
        borderRadius: '12px',
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        whiteSpace: 'nowrap',
      }}
    />
  </Box>
)

export const ButtonIcon = ({ icon, variant, text, badgeValue = 0, badgeVariant = 'primary', maxWidth, ...props }) => {
  let fontSize
  if (Array.isArray(variant)) {
    const indexes = variant.map(v => Object.keys(theme.buttons).indexOf(v))
    indexes.map(index => {
      if (Object.values(theme.buttons)[index].fontSize !== undefined) {
        fontSize = Object.values(theme.buttons)[index].fontSize
      }
      return fontSize
    })
  } else {
    const index = Object.keys(theme.buttons).indexOf(variant)
    fontSize = Object.values(theme.buttons)[index].fontSize
  }

  return (
    <Box sx={ maxWidth ? { position: 'relative', width: '100%' } : { position: 'relative' }}>
      <Box
        as='button'
        tx='buttons'
        variant={variant}
        {...props}
        __css={{
          appearance: 'none',
          display: 'flex',
          lineHeight: 'inherit',
          fontFamily: 'Nunito Sans',
          fontWeight: 'bold',
          textAlign: 'center',
          textDecoration: 'none',
          cursor: 'pointer',
          outline: 'none',
          border: 0,
          flexDirection: 'row',
          justifyContent: 'center',
          alignItems: 'center',
          whiteSpace: 'nowrap',
          position: 'relative',
        }}
      >
        { icon && <Icon icon={icon} fontSize={fontSize}/> }
        { badgeValue !== 0 && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', top: '2px', left: '16px' }}>{ badgeValue }</Badge> }
        { text && <span>{text}</span> }
      </Box>
    </Box>
  )
}

export const ButtonHoldPress = ({ variant, active = false, text, icon, badgeValue = 0, badgeVariant = 'primary', isCollapsible, hasCheckbox = false, disabled = false, maxWidth, ...props }) => {
  let colorValue
  let colorValueIcon
  let fontSize
  let fontWeight
  if (Array.isArray(variant)) {
    const indexes = variant.map(v => Object.keys(theme.buttons).indexOf(v))
    indexes.map(index => {
      if (Object.values(theme.buttons)[index][':focus'] !== undefined && active) {
        colorValue = Object.values(theme.buttons)[index][':focus'].color
        fontWeight = Object.values(theme.buttons)[index][':focus'].fontWeight
        if (Object.values(theme.buttons)[index].colorIcon !== undefined) {
          colorValueIcon = Object.values(theme.buttons)[index][':focus'].colorIcon
        }
      } else if (Object.values(theme.buttons)[index].color !== undefined) {
        colorValue = Object.values(theme.buttons)[index].color
      } else if (Object.values(theme.buttons)[index].colorIcon !== undefined) {
        colorValueIcon = Object.values(theme.buttons)[index].colorIcon
      } else if (Object.values(theme.buttons)[index].fontWeight !== undefined) {
        fontWeight = Object.values(theme.buttons)[index].fontWeight
      }
      if (Object.values(theme.buttons)[index].fontSize !== undefined) {
        fontSize = Object.values(theme.buttons)[index].fontSize
      }
      return colorValue
    })
  } else {
    const index = Object.keys(theme.buttons).indexOf(variant)
    colorValue = Object.values(theme.buttons)[index].color
    colorValueIcon = Object.values(theme.buttons)[index].colorIcon
    fontSize = Object.values(theme.buttons)[index].fontSize
    fontWeight = Object.values(theme.buttons)[index].fontWeight
    if (active) {
      colorValue = Object.values(theme.buttons)[index][':focus'].color
      colorValueIcon = Object.values(theme.buttons)[index][':focus'].colorIcon
      fontWeight = Object.values(theme.buttons)[index][':focus'].fontWeight
    }
  }

  return (
    <Box sx={ maxWidth ? { position: 'relative', width: '100%' } : { position: 'relative' }}>
      <Box
        as='button'
        tx='buttons'
        variant={variant}
        {...props}
        __css={{
          appearance: 'none',
          display: 'flex',
          lineHeight: 'inherit',
          fontFamily: 'Nunito Sans',
          fontWeight: {fontWeight},
          textAlign: 'center',
          textDecoration: 'none',
          cursor: 'pointer',
          outline: 'none',
          border: 0,
          flexDirection: 'row',
          //  justifyContent: 'center',
          width: maxWidth &&'100%',
          alignItems: 'center',
          whiteSpace: 'nowrap',
        }}
      >
        { hasCheckbox && active && <Checkbox defaultChecked disabled={disabled}/>}
        { hasCheckbox && !active && <Checkbox defaultChecked={false} disabled={disabled}/>}
        { icon && <Icon icon={icon} fontSize={fontSize} style={{ color: ( colorValueIcon || colorValue ) }}/> }
        { text && <span style={colorValue && { color: colorValue, fontWeight: fontWeight }}>{text}</span>}
        { badgeValue === 0 && isCollapsible && <span className='icon-dropdown' style={{ position: 'absolute', right: '10px', color: '#798B97', fontSize: '14px', transform: active ? 'rotate(-180deg)' : 'rotate(0deg)', paddingTop: '4px' }}/> }
        { badgeValue !== 0 && !text && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', top: '2px', left: '16px' }}>{badgeValue}</Badge> }
        { badgeValue !== 0 && text && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', right: '16px' }}>{badgeValue}</Badge> }
      </Box>
    </Box>
  )
}

export const ButtonUser = ({ variant, srcImage, text, ...props }) => (
  <Box sx={{ position: 'relative' }}>
    <Box
      as='button'
      tx='buttons'
      variant={variant}
      {...props}
      __css={{
        appearance: 'none',
        display: 'flex',
        lineHeight: 'inherit',
        fontFamily: 'Nunito Sans',
        fontWeight: 'bold',
        fontSize: '14px',
        textAlign: 'center',
        textDecoration: 'none',
        cursor: 'pointer',
        outline: 'none',
        color: 'white',
        bg: 'primary',
        border: 0,
        borderRadius: '12px',
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        height: '48px',
        whiteSpace: 'nowrap',
        padding: '13px 26px 13px 26px',
        gap: '8px',
      }}
    >
      {srcImage && <Image src={srcImage} variant="avatar"/> }
      {!srcImage && <span className='icon-contacto' style={{ fontSize: '24px', height: '24px' }}/>}
      {text}
    </Box>
  </Box>
)
