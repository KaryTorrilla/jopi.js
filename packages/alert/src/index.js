import React from 'react'
import { Flex } from '@oneloop/box'
import { Check, Exclamation, Times } from '@oneloop/icons'
import { Text } from '@oneloop/text'
import { Button } from '@oneloop/button'

export const Alert = ({ type = 'success', text, ...props }) => (
  <Flex
    variant="alert"
    {...props}
    __css={{
      width: '100%',
      bg: 'white',
      borderRadius: 'circle',
      boxShadow: 'active',
      justifyContent: 'space-between',
      p: 1,
    }}
  >
    {type === 'success' && (
      <Check
        bg="success"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    {type === 'warning' && (
      <Exclamation
        bg="warning"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    {type === 'danger' && (
      <Times
        bg="danger"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    <Text variant="body.0" display="flex" alignItems="center" flexGrow={1}>
      {text}
    </Text>
    <Button
      size={32}
      bg="#F6F2F2"
      p={3}
      display="inline-flex"
      alignSelf="center"
      mr={2}
    >
      <Times color="#716F6F" m="auto" />
    </Button>
  </Flex>
)
