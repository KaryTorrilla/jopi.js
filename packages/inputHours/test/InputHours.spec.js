import React from 'react'
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import 'jest-styled-components'
import { InputHours } from '../src'

Enzyme.configure({ adapter: new Adapter() })

const defaultHours = []

for (let i = 0; i < 24; i++) {
  for (let j = 0; j < 4; j++) {
    defaultHours.push(`${i < 10 ? '0' + i : i}:${j === 0 ? '00' : 15 * j}`)
  }
}

describe('InputHours', () => {
  it('renders correctly', () => {
    const component = shallow(
      <InputHours
        inputTime={new Date()}
        arrayInput={defaultHours}
      />
    )

    expect(component).toMatchSnapshot()
  })

  it('show dropdown', () => {
    const component = mount(
      <InputHours
        inputTime={new Date()}
        arrayInput={defaultHours}
      />
    )
    const listWrapper = component.find('.list')

    expect(listWrapper).toMatchSnapshot()
  })

  it('change hour', () => {
    const component = mount(
      <InputHours
        inputTime={new Date()}
        arrayInput={defaultHours}
      />
    )
    const input = component.find('.inputHourDisplay').at(1)
    const item = component.find('li').at(5)
    item.simulate('click')
    input
      .simulate('change', { target: { value: '00' } })
      .simulate('keyDown', { key: 'Enter' })
      .simulate('keyDown', { key: 'ArrowUp' })
      .simulate('keyDown', { key: 'ArrowDown' })
  })
})
