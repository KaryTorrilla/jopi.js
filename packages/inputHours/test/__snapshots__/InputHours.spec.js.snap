// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputHours renders correctly 1`] = `ShallowWrapper {}`;

exports[`InputHours show dropdown 1`] = `ReactWrapper {}`;
